src/snowflake/snowpark/dataframe_reader.py

```diff
@@ -452,3 +472,30 @@ def _read_semi_structured_file(self, path: str, format: str) -> DataFrame:
         )
         df._reader = self
         return df
+
+    def _upload_local_file_to_stage(self, path: str, format: str) -> str:
+        temp_stage = self._session.get_session_stage()
+        stage_path = path
+        if not path.startswith(STAGE_PREFIX) and not os.path.exists(path):
+            raise ValueError(f"The local file {path} does not exist")
+        else:
+            if os.path.isfile(path):
+                self._session.file.put(
+                    path, temp_stage, auto_compress=False, overwrite=True
+                )
+                _, filename = os.path.split(path)
+                stage_path = f"{temp_stage}/{filename}"
+            if os.path.isdir(path):
+                if format != "csv":
+                    raise ValueError(
+                        f"Only support a local directory for reading CSV files, but got {format}"
+                    )
+                filelist = os.listdir(path)
+                for file in filelist:
+                    if os.path.splitext(file)[-1][1:].lower() == format.lower():
+                        filepath = os.path.join(path, file)
+                        self._session.file.put(
+                            filepath, temp_stage, auto_compress=False, overwrite=True
+                        )
```

Comment:
`file.put` supports pattern. Can we use that? I guess the performance is better than uploading files one by one.

Reply:
I will revise this